internal/stack/serverless.go

```diff
@@ -0,0 +1,462 @@
+		err = project.CreateAgentPolicy(options.StackVersion)
+		if err != nil {
+			return fmt.Errorf("failed to create agent policy: %w", err)
+		}
+
+		// logger.Infof("Replacing GeoIP databases")
+		// err = cp.replaceGeoIPDatabases(config, options, settings.TemplateID, settings.Region, payload.Resources.Elasticsearch[0].Plan.ClusterTopology)
+		// if err != nil {
+		// 	return fmt.Errorf("failed to replace GeoIP databases: %w", err)
+		// }
+	case nil:
+		logger.Debugf("%s project existed: %s", project.Type, project.Name)
+		printUserConfig(options.Printer, config)
+		// logger.Infof("Updating project %s", project.Name)
+		// err = sp.updateDeployment(project, settings)
+		// if err != nil {
+		// 	return fmt.Errorf("failed to update deployment: %w", err)
+		// }
+	}
+
+	logger.Infof("Starting local agent")
+	err = sp.startLocalAgent(options, config)
+	if err != nil {
+		return fmt.Errorf("failed to start local agent: %w", err)
+	}
+
+	return nil
+}
+
+func (sp *serverlessProvider) composeProjectName() string {
+	return DockerComposeProjectName(sp.profile)
+}
+
+func (sp *serverlessProvider) localAgentComposeProject() (*compose.Project, error) {
+	composeFile := sp.profile.Path(profileStackPath, ServerlessComposeFile)
+	return compose.NewProject(sp.composeProjectName(), composeFile)
+}
+
+func (sp *serverlessProvider) startLocalAgent(options Options, config Config) error {
+	err := applyServerlessResources(sp.profile, options.StackVersion, config)
+	if err != nil {
+		return fmt.Errorf("could not initialize compose files for local agent: %w", err)
+	}
+
+	project, err := sp.localAgentComposeProject()
+	if err != nil {
+		return fmt.Errorf("could not initialize local agent compose project")
+	}
+
+	err = project.Build(compose.CommandOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to build images for local agent: %w", err)
+	}
+
+	err = project.Up(compose.CommandOptions{ExtraArgs: []string{"-d"}})
+	if err != nil {
+		return fmt.Errorf("failed to start local agent: %w", err)
+	}
+
+	return nil
+}
+
+func (sp *serverlessProvider) TearDown(options Options) error {
+	config, err := LoadConfig(sp.profile)
+	if err != nil {
+		return fmt.Errorf("failed to load configuration: %w", err)
+	}
+
+	err = sp.destroyLocalAgent()
+	if err != nil {
+		return fmt.Errorf("failed to destroy local agent: %w", err)
+	}
+
+	project, err := sp.currentProject(config)
+	if err != nil {
+		return fmt.Errorf("failed to find current project: %w", err)
+	}
+
+	logger.Debugf("Deleting project %q (%s)", project.Name, project.ID)
+
+	err = sp.deleteProject(project, options)
+	if err != nil {
+		return fmt.Errorf("failed to delete project: %w", err)
+	}
+
+	// logger.Debugf("Deleting GeoIP bundle.")
+	// err = cp.deleteGeoIPExtension()
+	// if err != nil {
+	// 	return fmt.Errorf("failed to delete GeoIP extension: %w", err)
+	// }
+
+	// err = storeConfig(sp.profile, Config{})
+	// if err != nil {
+	// 	return fmt.Errorf("failed to store config: %w", err)
+	// }
```

Comment:
This is needed, right?

Reply:
This was an attempt to clean the profile config.
But if this is done, then next `elastic-package stack up` executions would need to provide always `--provider serverless`

Currently, compose provider does not do this cleanup neither.

I think these can be removed, WDYT @jsoriano?